pkg/kmodule/kmodule_linux.go

```diff
@@ -33,17 +33,52 @@ const (
 	MODULE_INIT_IGNORE_VERMAGIC = 0x2
 )
 
+type LinuxLoader struct {
+	modules string
+}
+
 // Init loads the kernel module given by image with the given options.
-func Init(image []byte, opts string) error {
+func (l *LinuxLoader) Init(image []byte, opts string) error {
 	return unix.InitModule(image, opts)
 }
 
+func (l *LinuxLoader) List(w io.Writer) error {
+	file, err := os.Open(l.modules)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+
+	if _, err := fmt.Fprintln(w, "Module                  Size  Used by"); err != nil {
+		return err
+	}
+
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		s := strings.Split(scanner.Text(), " ")
```

Comment:
Perhaps [`strings.Fields`](https://pkg.go.dev/strings#Fields), which splits on "one or more consecutive white space characters", instead of `strings.Split(x, " ")`?

Reply:
done